src/cloudwatch-applicationsignals-mcp-server/awslabs/cloudwatch_applicationsignals_mcp_server/enablement_guides/templates/ec2/ec2-dotnet-enablement.md

```diff
@@ -0,0 +1,459 @@
+  "cat > /opt/aws/amazon-cloudwatch-agent/etc/amazon-cloudwatch-agent.json << 'EOF'",
+  '{',
+  '  "traces": {',
+  '    "traces_collected": {',
+  '      "application_signals": {}',
+  '    }',
+  '  },',
+  '  "logs": {',
+  '    "metrics_collected": {',
+  '      "application_signals": {}',
+  '    }',
+  '  }',
+  '}',
+  'EOF',
+  '',
+  '# Start CloudWatch Agent with Application Signals configuration',
+  '/opt/aws/amazon-cloudwatch-agent/bin/amazon-cloudwatch-agent-ctl \\',
+  '  -a fetch-config \\',
+  '  -m ec2 \\',
+  '  -s \\',
+  '  -c file:/opt/aws/amazon-cloudwatch-agent/etc/amazon-cloudwatch-agent.json',
+);
+```
+
+**For Windows Server instances:**
+
+**CDK TypeScript example:**
+```typescript
+instance.userData.addCommands(
+  '# Create CloudWatch Agent configuration for Application Signals',
+  '@"',
+  '{',
+  '  "traces": {',
+  '    "traces_collected": {',
+  '      "application_signals": {}',
+  '    }',
+  '  },',
+  '  "logs": {',
+  '    "metrics_collected": {',
+  '      "application_signals": {}',
+  '    }',
+  '  }',
+  '}',
+  '"@ | Out-File -FilePath "C:\\ProgramData\\Amazon\\AmazonCloudWatchAgent\\amazon-cloudwatch-agent.json" -Encoding utf8',
+  '',
+  '# Start CloudWatch Agent with Application Signals configuration',
+  '& "C:\\Program Files\\Amazon\\AmazonCloudWatchAgent\\amazon-cloudwatch-agent-ctl.ps1" -a fetch-config -m ec2 -s -c file:"C:\\ProgramData\\Amazon\\AmazonCloudWatchAgent\\amazon-cloudwatch-agent.json"',
+);
+```
+
+### Step 6: Install ADOT .NET Auto-Instrumentation
+
+Choose based on deployment type and OS identified in "Before You Start".
+
+#### Option A: Docker Deployment - Modify Dockerfile
+
+For Docker deployments, modify the `Dockerfile` in the application directory.
+
+**For Linux-based containers:**
+
+Add these lines to install the ADOT .NET auto-instrumentation. Place this AFTER any `RUN` commands that install dependencies, but BEFORE the `CMD` line:
+
+```dockerfile
+# Install unzip (required by ADOT installation script)
+# Use the appropriate package manager for your base image:
+# - For Amazon Linux 2: RUN yum install -y unzip
+# - For Amazon Linux 2023: RUN dnf install -y unzip
+# - For Ubuntu/Debian: RUN apt-get update && apt-get install -y unzip
+RUN dnf install -y unzip  # Adjust package manager as needed
+
+# Download and install ADOT .NET auto-instrumentation to /opt (accessible by all users)
+RUN curl -L -O https://github.com/aws-observability/aws-otel-dotnet-instrumentation/releases/latest/download/aws-otel-dotnet-install.sh \
+    && chmod +x ./aws-otel-dotnet-install.sh \
+    && OTEL_DOTNET_AUTO_HOME="/opt/otel-dotnet-auto" ./aws-otel-dotnet-install.sh \
+    && chmod -R 755 /opt/otel-dotnet-auto
+```
+
+**For Windows-based containers:**
+
+```dockerfile
+# Download and install ADOT .NET auto-instrumentation
+RUN $module_url = "https://github.com/aws-observability/aws-otel-dotnet-instrumentation/releases/latest/download/AWS.Otel.DotNet.Auto.psm1"; \
```

Comment:
The base image of Windows container may not provide PowerShell. For example, [nanoserver](https://hub.docker.com/r/microsoft/windows-nanoserver) only provides cmd.exe as shell.